src/Nethermind/Nethermind.Core/TypeDiscovery.cs

```diff
@@ -124,20 +130,32 @@ static bool Filter(Dictionary<string, Assembly> considered, AssemblyName an)
 
         public static IEnumerable<Type> FindNethermindTypes(Type baseType)
         {
-            LoadAll();
+            Initialize();
+
+            return FindNethermindTypes(_nethermindAssemblies, baseType);
+        }
+
+        private static IEnumerable<Type> FindNethermindTypes(IEnumerable<Assembly> assemblies, Type baseType)
+        {
+            Func<Assembly, IEnumerable<Type>> assembliesSelector = a => FindNethermindTypes(a, baseType);
 
-            return _nethermindAssemblies
-                .SelectMany(a => (a?.IsDynamic ?? false ? Array.Empty<Type>() : a?.GetExportedTypes())?
-                    .Where(t => baseType.IsAssignableFrom(t) && baseType != t) ?? Array.Empty<Type>());
+            return assemblies.SelectMany(assembliesSelector);
         }
 
+        private static IEnumerable<Type> FindNethermindTypes(Assembly assembly, Type baseType) =>
```

Comment:
Shouldn't this method be renamed since now it doesn't necessarily return only Nethermind types?

Reply:
Well it does, but those Nethermind types (IConfig, Metrics, INethermindPlugin) can be by 3rd parties? But TBH don't care, do you have better name suggestion?